### PR TITLE
Add new logic to safety tree

### PR DESCRIPTION
**Pull request by pkowalsk1**
`bump::patch`

Battery health checks were added in this pull request.

### CODE REVIEW

File: panther_manager/config/safety.xml [hunk 1]
```diff
@@ -2,45 +2,72 @@
...
   <BehaviorTree ID="Safety">
     <Sequence>
-      <Sequence name="BatteryStatusSequence"
-                _skipIf="battery_health == POWER_SUPPLY_HEALTH_GOOD \
-|| battery_health == POWER_SUPPLY_HEALTH_UNKNOWN">
+      <Sequence name="BatteryOverheatSequence"
+                _skipIf="battery_health != POWER_SUPPLY_HEALTH_OVERHEAT">
+        <Sequence name="FatalBatTempSequence"
+                  _skipIf="bat_temp &lt;= FATAL_BAT_TEMP">
+          <SignalShutdown reason="Battery status unhealthy"/>
```
Comment (KmakD) [inline panther_manager/config/safety.xml]:
```suggestion
          <SignalShutdown reason="Fatal battery temperature"/>
```

File: panther_manager/config/safety.xml [hunk 2]
```diff
@@ -2,45 +2,72 @@
...
+                  _skipIf="fan_state">
+          <CallSetBoolService name="EnableFanIfHighBatTemp"
+                              data="true"
+                              service_name="hardware/fan_enable"
+                              timeout="100"/>
+        </Sequence>
+      </Sequence>
+      <Sequence name="BatteryDeadSequence"
+                _skipIf="battery_health != POWER_SUPPLY_HEALTH_DEAD">
         <SignalShutdown reason="Battery status unhealthy"/>
```
Comment (KmakD) [inline panther_manager/config/safety.xml]:
```suggestion
        <SignalShutdown reason="Battery dead"/>
```

File: panther_manager/config/safety.xml [hunk 3]
```diff
@@ -2,45 +2,72 @@
...
+        <Sequence name="FatalBatTempSequence"
+                  _skipIf="bat_temp &lt;= FATAL_BAT_TEMP">
+          <SignalShutdown reason="Battery status unhealthy"/>
+        </Sequence>
+        <Sequence name="CriticalBatTempSequence"
+                  _skipIf="bat_temp &lt;= CRITICAL_BAT_TEMP">
+          <CallTriggerService name="TriggerEStop"
+                              service_name="hardware/e_stop_trigger"
+                              timeout="100"
+                              _skipIf="e_stop_state"/>
```
Comment (KmakD) [inline panther_manager/config/safety.xml:16]: Should e-stop be triggered when critical battery temp is reached or when battery signals overheat?

File: panther_manager/config/safety.xml [hunk 4]
```diff
@@ -2,45 +2,72 @@
...
+          <CallSetBoolService name="DisableAUX"
+                              data="false"
+                              service_name="hardware/aux_power_enable"
+                              timeout="100"
+                              _skipIf="!aux_state || bat_temp &lt;= CRITICAL_BAT_TEMP"/>
```
Comment (KmakD) [inline panther_manager/config/safety.xml]:
```suggestion
                              _skipIf="!aux_state"/>
```
That condition is already checked by the sequence.

File: panther_manager/config/safety.xml [hunk 5]
```diff
@@ -2,45 +2,72 @@
...
+        </Sequence>
+        <Sequence name="HighBatTempSequence"
+                  _skipIf="fan_state">
+          <CallSetBoolService name="EnableFanIfHighBatTemp"
+                              data="true"
+                              service_name="hardware/fan_enable"
+                              timeout="100"/>
+        </Sequence>
```
Comment (KmakD) [inline panther_manager/config/safety.xml]: I think skip if condition should be moved directly to EnableFanIfHighBatTemp. It refers to fan enabling only. HighBatTempSequence may be extended with other behaviors.

**Review by KmakD — CHANGES_REQUESTED** [review]

File: panther_manager/config/safety.xml (hunk 3 above)
Comment (pkowalsk1) [reply panther_manager/config/safety.xml:16]: According to the arrangement with Radek, should be triggered when critical battery temp is reached

File: panther_manager/config/safety.xml [hunk 6]
```diff
@@ -2,45 +2,72 @@
...
+                              timeout="100"
+                              _skipIf="!aux_state"/>
+        </Sequence>
+        <Sequence name="HighBatTempSequence">
+          <CallSetBoolService name="EnableFanIfHighBatTemp"
+                              data="true"
+                              service_name="hardware/fan_enable"
+                              timeout="100"
+                              _skipIf="fan_state"/>
+        </Sequence>
```
Comment (KmakD) [inline panther_manager/config/safety.xml]: Actually,  HighBatTempSequence can be removed as all its behaviors can be encapsulated in BatteryOverheatSequence.

**Review by KmakD — CHANGES_REQUESTED** [review]

**Review by KmakD — APPROVED** [review]

**Pull request merged**